### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by IronGauntlets**
Please see the commit history.

### CODE REVIEW

File: rpc/handlers.go [hunk 1]
```diff
@@ -182,34 +182,35 @@ func adaptDeclareTransaction(t *core.DeclareTransaction) *Transaction {
 
 	if t.Version.Equal(new(felt.Felt).SetUint64(2)) {
 		txn.CompiledClassHash = nil // todo: add when we have support for Declare V2
-	} else if !t.Version.IsZero() && !t.Version.IsOne() {
```
Comment (wojciechos) [inline rpc/handlers.go:185]: Was the removal of this specific 'else if' intentional?

**Review by wojciechos — APPROVED** [review]

File: rpc/handlers.go [hunk 2]
```diff
@@ -182,34 +182,35 @@ func adaptDeclareTransaction(t *core.DeclareTransaction) *Transaction {
...
 	}
+
 	return txn
 }
 
 func (h *Handler) blockByID(id *BlockID) (*core.Block, error) {
 	if id.Latest {
 		return h.bcReader.Head()
-	} else if id.Hash != nil {
+	}
```
Comment (omerfirmak) [inline rpc/handlers.go:193]: Let's make this a `switch`? The conditions we are testing are mutually exclusive, so I believe we should reflect that it code.

File: rpc/handlers.go [hunk 3]
```diff
@@ -182,34 +182,35 @@ func adaptDeclareTransaction(t *core.DeclareTransaction) *Transaction {
...
-		return h.bcReader.BlockByNumber(id.Number)
 	}
+	return h.bcReader.BlockByNumber(id.Number)
 }
 
 func (h *Handler) blockHeaderByID(id *BlockID) (*core.Header, error) {
 	if id.Latest {
 		return h.bcReader.HeadsHeader()
-	} else if id.Hash != nil {
+	}
```
Comment (omerfirmak) [inline rpc/handlers.go:206]: this as well

File: starknetdata/feeder/feeder.go [hunk 4]
```diff
@@ -267,11 +267,13 @@ func (f *Feeder) Class(ctx context.Context, classHash *felt.Felt) (core.Class, e
 
 	if response.V1 != nil {
 		return adaptCairo1Class(response.V1)
-	} else if response.V0 != nil {
+	}
+
+	if response.V0 != nil {
```
Comment (omerfirmak) [inline starknetdata/feeder/feeder.go:272]: this as well

**Review by omerfirmak — CHANGES_REQUESTED** [review]

File: rpc/handlers.go (hunk 1 above)
Comment (IronGauntlets) [reply rpc/handlers.go:185]: Yes since we trust the source of information.

File: rpc/handlers.go (hunk 2 above)
Comment (IronGauntlets) [reply rpc/handlers.go]: done

File: rpc/handlers.go (hunk 3 above)
Comment (IronGauntlets) [reply rpc/handlers.go]: done

File: starknetdata/feeder/feeder.go (hunk 4 above)
Comment (IronGauntlets) [reply starknetdata/feeder/feeder.go]: done

**Review by omerfirmak — APPROVED** [review]

**Pull request merged**